bot/exts/moderation/voice_gate.py

```diff
@@ -0,0 +1,136 @@
+NOT_ENOUGH_DAYS_AFTER_VERIFICATION = f"haven't been verified for at least {VoiceGateConf.minimum_days_verified} days"
+VOICE_BANNED = "are voice banned"
+
+FAILED_MESSAGE = """{user} you don't meet with our current requirements to pass Voice Gate. You {reasons}."""
+
+
+class VoiceGate(Cog):
+    """Voice channels verification management."""
+
+    def __init__(self, bot: Bot):
+        self.bot = bot
+
+    @property
+    def mod_log(self) -> ModLog:
+        """Get the currently loaded ModLog cog instance."""
+        return self.bot.get_cog("ModLog")
+
+    @command(aliases=('voiceverify', 'vverify', 'voicev', 'vv'))
+    @has_no_roles(Roles.voice_verified)
+    @in_whitelist(channels=(Channels.voice_gate,), redirect=None)
+    async def voice_verify(self, ctx: Context, *_) -> None:
+        """
+        Apply to be able to use voice within the Discord server.
+
+        In order to use voice you must meet all three of the following criteria:
+        - You must have over a certain number of messages within the Discord server
+        - You must have accepted our rules over a certain number of days ago
+        - You must not be actively banned from using our voice channels
+        """
+        try:
+            data = await self.bot.api_client.get(f"bot/users/{ctx.author.id}/metricity_data")
+        except ResponseCodeError as e:
+            if e.status == 404:
+                await ctx.send(f":x: {ctx.author.mention} Unable to find Metricity data about you.")
+                log.info(f"Unable to find Metricity data about {ctx.author} ({ctx.author.id})")
+            else:
+                log.warning(f"Got response code {e.status} while trying to get {ctx.author.id} metricity data.")
+                await ctx.send(":x: Got unexpected response from site. Please let us know about this.")
+            return
+
+        # Pre-parse this for better code style
+        data["verified_at"] = parser.isoparse(data["verified_at"])
+
+        failed = False
+        failed_reasons = []
+
+        if data["verified_at"] > datetime.utcnow() - timedelta(days=VoiceGateConf.minimum_days_verified):
+            failed_reasons.append(NOT_ENOUGH_DAYS_AFTER_VERIFICATION)
+            failed = True
+            self.bot.stats.incr("voice_gate.failed.verified_at")
+        if data["total_messages"] < VoiceGateConf.minimum_messages:
+            failed_reasons.append(NOT_ENOUGH_MESSAGES)
+            failed = True
+            self.bot.stats.incr("voice_gate.failed.total_messages")
+        if data["voice_banned"]:
+            failed_reasons.append(VOICE_BANNED)
+            failed = True
+            self.bot.stats.incr("voice_gate.failed.voice_banned")
+
+        if failed:
+            if len(failed_reasons) > 1:
+                reasons = f"{', '.join(failed_reasons[:-1])} and {failed_reasons[-1]}"
+            else:
+                reasons = failed_reasons[0]
+
+            await ctx.send(
```

Comment:
Tempted to say that for clarity we should DM the user with this instead. The message disappearing makes it difficult to read what the conditions are while testing locally, so I can't imagine it being too clear in production.

If we want some feedback in the channel, we could have a "check your DMs" message in the channel.
```suggestion
            await ctx.author.send(
```